client/state/sites/domains/reducer.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Internal dependencies
+ */
+import {
+	SITE_DOMAINS_FETCH,
+	SITE_DOMAINS_FETCH_COMPLETED,
+	SITE_DOMAINS_FETCH_FAILED,
+	SERIALIZE,
+	DESERIALIZE
+} from 'state/action-types';
+
+export const initialSiteState = {
+	data: null,
+	error: null,
+	hasLoadedFromServer: false,
+	isRequesting: false
+};
```

Comment:
If we want to persist domains in the future I'd suggest splitting out temporary data into different subtrees.
```text
state.sites.domains.items,
state.sites.domains.requesting,
state.sites.domains.errors
```
See `state/post-types/reducer` for an example of this